services/business/checkers/route_rules/route_checker.go

```diff
@@ -0,0 +1,58 @@
+package route_rules
+
+import (
+	"reflect"
+	"strconv"
+
+	"github.com/kiali/kiali/kubernetes"
+	"github.com/kiali/kiali/services/models"
+)
+
+type RouteChecker struct{ kubernetes.IstioObject }
+
+func (route RouteChecker) Check() ([]*models.IstioCheck, bool) {
+	validations := make([]*models.IstioCheck, 0)
+
+	var weightSum uint64
+	var weightCount int
+	var valid bool = true
+
+	slice := reflect.ValueOf(route.GetSpec()["route"])
+	if slice.Kind() != reflect.Slice {
+		return validations, valid
+	}
+
+	for i := 0; i < slice.Len(); i++ {
+		var weight uint64
+
+		route := slice.Index(i).Interface().(map[string]interface{})
+		if route["weight"] == nil {
+			continue
+		}
+
+		weightCount = weightCount + 1
+		weight = route["weight"].(uint64)
```

Comment:
If weight is wrongly defined with a string, this is going to panic, isn't it ?

Reply:
Ah, no, it's a map of string.

Reply:
Well, no, my previous comment is valid, as route is a map[string]interface{}, so  route["weight"].(uint64) perhaps may fail (imagine that a user put a 98% and converts the 98 number in a 98% string).

Reply:
yes, you are right. It just panics. uint64 is an unsignedint, so anything different than that just breaks.